### ISSUE TITLE
NaN during training

### ISSUE THREAD

**Issue by jankyee** [irfanICMLL/structure_knowledge_distillation#21]
HI, I try to reproduce the training process, but get NaN D_loss after several training steps. I think that is because `torch.sqrt` in `CriterionAdditionalGP`  rerurns a NaN. I refer to https://github.com/pytorch/pytorch/issues/2534 and add 1e-8 to the grad_norm, 

    
![image](https://user-images.githubusercontent.com/21698398/72490168-6bc80b00-3851-11ea-9752-5043af5ec574.png)

but can't help. Any suggestions?

**Comment by jankyee** [comment]
I find that `grad_l2_sum`  may explode after several steps
![image](https://user-images.githubusercontent.com/21698398/72491271-818aff80-3854-11ea-9501-0e66a61a3df8.png)

so, adding 1e-8 can't help. But why exploding...

**Comment by irfanICMLL** [comment]
Have you modified the training code or the config of the project?

**Comment by jankyee** [comment]
> Have you modified the training code or the config of the project?

 I just modified the dataset and pretrain path. 

Now, I try using `adv-loss-type: hinge`, and have trained 1300 steps, until now everything seems ok...

**Comment by irfanICMLL** [comment]
Ok, the hinge loss can also get similar performance.

**Comment by jankyee** [comment]
> Ok, the hinge loss can also get similar performance.

emmm...the reproduced result seems worse. I sent you my training log, would you kindly help reviewing my settings? Or share your logs?

**Comment by irfanICMLL** [comment]
I have already sent the training log to you through email. What data are you using? Have you obtained a good teacher?

**Comment by sungsooo** [comment]
@Jankyee 
Hi, I got the same error!
Did you resolve this problem?
If so, could you help me?
Thanks.

**Comment by songyang86** [comment]
can you sent your training log to me? I also got  worse result
